pkg/client/client.go

```diff
@@ -301,9 +304,20 @@ func (tc *TelemetryClient) Register() (err error) {
 }
 
 func (tc *TelemetryClient) Generate(telemetry types.TelemetryType, content []byte, tags types.Tags) error {
+	log.Printf("Generated Telemetry:\nName: %q\nTags: %v\nContent: %s\nSize: %v\n",
+		telemetry, tags, content, utils.HumanReadableSize(content))
+
+	// Enforce size limits
+	tl := &telemetrylib.TelemetryDataLimits{}
```

Comment:
Instead of setting things up here I would define a NewTelemetryDataLimits() routine which returns a pointer to a TelemetryDataLimits struct which has had the limits already set.

Optionally define an Init() method for the struct pointer that does the initialisation, and have the NewTelemetryDataLimits() routine use the Init() method to do the initialisation.

Reply:
This should all be in the limits.go file itself, with the lib.NewTelemetryDataLimits() call here to obtain a limits struct pointer.